instructure/canvas-ios
# Background sync scheduler

### Notes
1. Wifi only sync switch logic will be implemented in another ticket.
2. The case when the offline sync is disabled on OS level is not handled yet and will be improved in a follow-up ticket as well.

refs: MBL-16758
affects: Student
release note: none

test plan: 

### Debug Testing
1. Build the app to a real device.
3. Go to Offline Content Settings.
4. Enable auto sync.
5. Set sync frequency to "As frequent as the OS allows (DEBUG)".
6. Put the app into the background, this will schedule a sync.
7. Stop the app with the debugger and enter the following command: `e -l objc -- (void)[[BGTaskScheduler sharedScheduler] _simulateLaunchForTaskWithIdentifier:@"com.instructure.icanvas.offline-sync"]`
8. Continue running the app, sync should start.
9. To abort an in progress sync enter the following debugger command: `e -l objc -- (void)[[BGTaskScheduler sharedScheduler] _simulateExpirationForTaskWithIdentifier:@"com.instructure.icanvas.offline-sync"]`

### Manual Testing
1. Repeat steps 1-5 from Debug Testing steps. (Except that sync frequency have to be set to daily.)
2. Wait enough time for the sync to happen (mostly at night I assume).
3. Open the app, go to debug settings, check logs about the background sync.

## Checklist
